Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1975,9 +1976,22 @@ extension BrowserViewController: ContextMenuHelperDelegate {
                 dialogTitle = url.absoluteString
             }
 
+            let photoAuthorizeStatus = PHPhotoLibrary.authorizationStatus()
             let saveImageTitle = NSLocalizedString("Save Image", comment: "Context menu item for saving an image")
             let saveImageAction = UIAlertAction(title: saveImageTitle, style: UIAlertActionStyle.Default) { (action: UIAlertAction!) -> Void in
-                self.getImage(url) { UIImageWriteToSavedPhotosAlbum($0, nil, nil, nil) }
+                if photoAuthorizeStatus == PHAuthorizationStatus.Authorized || photoAuthorizeStatus == PHAuthorizationStatus.NotDetermined {
+                    self.getImage(url) { UIImageWriteToSavedPhotosAlbum($0, nil, nil, nil) }
```

Comment:
We should probably check to see if the saving photo caused an error. See https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIKitFunctionReference/#//apple_ref/c/func/UIImageWriteToSavedPhotosAlbum for the selector we need to implement for this. I don't think we need to do anything fancy except for tell the user that something went wrong and the image didn't save and to try again later.